prisondersdilemma.py

```diff
@@ -4,199 +4,438 @@
+			elif response == [c]: #participant is male caucasian 
+				opponent.setImage("opponent2.png")
+				participant.setImage("partC.png") 
+			elif response == [b]: #participant is female non-caucasian
+				opponent.setImage("opponent1.png") 
+				participant.setImage("partB.png")
+			else: #option d, participant is male non-caucasian
+				opponent.setImage("opponent1.png") 
+				participant.setImage("partD.png")
+			print('in-group condition')
+		
+		if condition == 2: #playing with an out-group member
+			if response == [a] #participant is female caucasian
+				opponent.setImage("opponent1.png")
+				participant.setImage("partA.png") 
+			elif response == [c]: #participant is male caucasian 
+				opponent.setImage("opponent1.png")
+				participant.setImage("partC.png") 
+			elif response == [b]: #participant is female non-caucasian
+				opponent.setImage("opponent2.png") 
+				participant.setImage("partB.png")
+			else: #option d, participant is male non-caucasian
+				opponent.setImage("opponent2.png") 
+				participant.setImage("partD.png")
+			print('out-group condition')
+		
+		if condition == 3: #playing with no information about opponent
+			opponent.setImage("anonymous.jpg")
+			if response == [a] #participant is female caucasian
+				participant.setImage("partA.png") 
+			elif response == [c]: #participant is male caucasian 
+				participant.setImage("partC.png") 
+			elif response == [b]: #participant is female non-caucasian
+				participant.setImage("partB.png")
+			else: #option d, participant is male non-caucasian
+				participant.setImage("partD.png")
+			print('anonymous condition')
+	
+	
+	
+	'''
+	Time to start the experiment!
+	'''
+	backgroundimage.draw #background prisonwall
+	introtxt.draw()
+	win.flip()
+	waitKeys(keyList = ('space')) #wait until participant pressed space to continue
+	
+	#choosing avatar that represents participant 
+	avattxt.draw()
+	partoverview.draw()
+	win.flip()
+	response = waitKeys(keyList = ('a', 'b','c','d'))
+	
+	shuffle(condlist) #randomly assigning condition (in group, out group, anonymous opponent)
+	condition = condlist[0]
+	avatarassignment()
+	
+		#waitkeys?
+	shuffle(strategy)
+	
+	#explaining the game
+	backgroundimage.draw()
+	insttxt.draw()
+	win.flip()
+	waitKeys(keyList = ('space')) #wait until participant read instructions and pressed space to start the trial loop
+	
+	#trial loop
+	background_image.draw
+	for i in range(ntrials):
 		
+	
+		#opponent always cooperates
+		if strategy == 1:
+			cooperation()
 			
-
-
-
-
-
-
-#exit: 
-if resp == ['q']:
-	break 
-
+		
+		#opponent always defects
+		elif strategy == 2:
+			defect()
+	
+		#random coop or defect
+		elif strategy == 3:
+			random()
+	
+		elif strategy == 4:
+			nicetittat()
+			
+		else strategy == 5:
+			susptittat()
+			
+			
+	
+	
+	#exit early: 
+		if resp == ['q']:
+			goodbye.draw()
+			win.flip()
+			wait(5)
+			break 
+	
+	goodbye.draw()
+	win.flip()
+	wait(5)
+	win.close()
+	log.close()
+	
```

Comment:
Kathi I wish I had more input to give you with this. It was clear and it's very evident you put a lot of thought and work into it. I saw some other syntax error but none that Dana hadn't given a solution to already. Perhaps some minor suggestions you could implement before Sunday. Good luck! You'll do great